### PR TITLE
Support for GDAL 2.2.x and merge of ogrext1/2

### PR DESCRIPTION
**Pull request by snorfalorpagus**
Following the comments in #461 this is an attempt to merge ogrext1 and ogrext2 into a single file, using the shim approach taken in rasterio.

### CODE REVIEW

File: fiona/_shim.pyx [hunk 1]
```diff
@@ -0,0 +1,55 @@
+cdef extern from "ogr_api.h":
```
Comment (snorfalorpagus) [inline fiona/_shim.pyx]: This file needs removing.

Comment (sgillies) [inline fiona/_shim.pyx]: I don't understand. Because it's taken care of elsewhere?

Comment (snorfalorpagus) [inline fiona/_shim.pyx]: Yes. I shouldn't have added it into the repo. It's a copy of either shim1, shim2 or shim22, in the same way that ogrext.pyx was a copy of ogrext1.pyx/ogrext2.pyx.

File: fiona/_shim.pyx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+from fiona.ogrext2 cimport *
+from fiona._err import cpl_errs
+
+cdef bint is_field_null(void *feature, int n):
+    if OGR_F_IsFieldNull(feature, n):
+        return True
+    elif not OGR_F_IsFieldSet(feature, n):
+        return True
+    else:
+        return False
```
Comment (sgillies) [inline fiona/_shim.pyx]: Great!

**Comment by snorfalorpagus** [comment]
I've been thinking about this issue some more. If we were to do a refactor like this, how do we keep up with the maint- branches? It becomes much more difficult to merge commits as the main files have moved around.

**Comment by sgillies** [comment]
@snorfalorpagus yes, it does become more difficult. The good news is that we have no more than 1 bug to fix in maint-1.7 and port to 1.8 and we don't have to maintain 1.7 once 1.8 is released since the latter is compatible.

**Comment by snorfalorpagus** [comment]
@sgillies I have some more time to look at this again now. I'll apply this approach to the master branch now, if that's OK?

**Comment by sgillies** [comment]
@snorfalorpagus yes please!

**Pull request closed without merging**